apps/api/src/course/interface/course.interface.ts

```diff
@@ -0,0 +1,25 @@
+import { Course, DayOfWeek, GenEdType, Period } from '@thinc-org/chula-courses'
+
+export interface ICourseSearchDocument {
+  abbrName: string
+  courseNo: string
+  courseNameTh: string
+  courseNameEn: string
+  courseDescTh: string
+  courseDescEn: string
+  genEdType: string
+  studyProgram: string
+  semester: string
+  academicYear: string
+  rawData: Course
+}
```

Comment:
the type is weird. It doesn't seem to match the type in GraphQL schema, no?

Reply:
This type is the schema of the document in Elasticsearch

Reply:
but this type is returned from a graphql resolver (search) which does NOT match its graphql type?